test/format/ops.c

```diff
@@ -523,6 +524,19 @@ prepare_transaction(TINFO *tinfo)
 
     lock_writeunlock(session, &g.ts_lock);
 
+    /*
+     * Sometimes add a delay after prepare to induce extra memory stress. For 80% of the threads,
+     * there is never a delay, so there is always a dedicated set of threads trying to do work. For
+     * the other 20%, we'll sometimes delay. For these threads, 99% of the time, proceed without
+     * delay. The rest of the time, pause up to 5 seconds, weighted toward the smaller delays.
+     */
+    if (tinfo->id % 5 == 0) {
+        longwait = mmrand(&tinfo->rnd, 0, 1000);
+        if (longwait < 10) {
+            pause_ms = mmrand(&tinfo->rnd, 0, 10) << longwait;
+            __wt_sleep(0, pause_ms * WT_THOUSAND);
+        }
+    }
```

Comment:
This seems reasonable to me, but the test against `tinfo.id` can be part of the `mmrand` test, can't it? That is, `tinfo.id % 5` is the same as 20%, can't we just do:
```
longwait = mmrand(&tinfo->rnd, 0, 1000);
if (longwait < 2)
```
Or maybe I'm just math-challenged, it was never my strong point.

And maybe there's an off-by-one, because the 0-1000 range is 1001 values, I didn't dig into it, I can't imagine it matters.

Reply:
About the mmrand test - I went this way first, but performance was much worse - there was more waiting.  Here's a way to think about it - if a thread can do 1000 ops/sec, and there is a 1% chance of having a wait on every op, then every thread will pause after working for 0.1 seconds, and sometimes for significant time.  The throughput may be half or less.  But if 80% of the threads are dedicated to work, no matter how slow the others are, throughput will be at least 80%.  Of course there may be secondary effects, but there's a real difference.

You're right about the off-by-one.  I'll fix it so it isn't confusing.